src/app/zap-templates/templates/app/CHIPClusters.zapt

```diff
@@ -1,6 +1,7 @@
 {{> header}}
 
-{{#if (chip_has_client_clusters)}}
+{{#all_user_clusters_with_outgoing_commands}}
```

Comment:
This is different from `chip_client_clusters` right ?  Looking at the output, it emit codes for the responses commands.

Reply:
Yes all_user_clusters_with_outgoing_commands is different from chip_has_client_clusters. I felt like we needed this for all the outgoing commands i.e. outgoing commands from client clusters as well as outgoing commands from server clusters. Correct me if I am wrong but the way I understood it is that the generated code was not handling outgoing commands from server to client. You are right that many of the server to client outgoing commands can be command responses but may be not all of them. Is there a reason we do not want this for command responses? Zigbee pro used to have something similar in the form of client-command-macro.h which used to generate for all outgoing commands disregarding the fact on whether the outgoing command was enabled or not by the user. This generated file seemed similar to that and based on the current strategy of just generating for enabled outgoing commands I added new helpers such that commands which are not enabled will not be generated for.

Reply:
There are no outgoing commands from server to client that we need codegen like this for.  All commands from server to client are responses in Matter.

In general, I would expect this PR to lead to no generated code changes except the cases where the current codegen is wrong (lighting-app, in particular).